### ISSUE TITLE
Doubts in Expansion of Abbreviation in Coding

### ISSUE THREAD

**Issue by PRANEETH-ALURU** [BruceWen120/medal#11]
Hello BruceWen!

I'm looking exclusively for abbreviations expansions in sentences and I came across your github repo.  I've few doubts from your work.

1) I started using the pre-trained models one by one starting with LSTM ( lstm = torch.hub.load("BruceWen120/medal", "lstm").
I want to disambiguate the abbreviations in top 10 sentences of test data downloaded from kaggle and later use my data. When I converted the test data to (batch=10 , sequence=230, dim=300) format using fast text on crawl-300d-2M-subword.bin  and gave input to the model, I got output of 22555 dim (as given) for every data point.  Can you tell me what is this output referring to ? What is the next step after getting 22555 dim outputs to get the expansions? From where and how can I get the expansions from these 22555 dim outputs?  

2) This tool of yours will disambiguate only single abbreviation in a sentence but not multiple as it location based.  So, it is mandatory to have locations in test data. Can you tell me how did you detected and extracted the locations of abbreviations from a given text ? Because, this has to be done on my text data also.

Expecting the reply soon...

**Comment by BruceWen120** [comment]
Thanks for your interest in this work!

1. I will need to check and get back to you in a few weeks about the 22555 output dimension. I was notified about this a while ago but never got the chance to dig into it.
2. For finding abbreviations you can check out [this script](https://github.com/BruceWen120/medal/blob/master/preprocess/batch_reverse_substitute.py). Essentially we have the mapping between abbreviations and expansions, and in documents we iterate over words and see if they match the expansions. About multiple abbreviations, that's doable too with the same logic.

**Comment by PRANEETH-ALURU** [comment]
Sure. Hoping to get the reply soon.

**Comment by phaniparsa** [comment]
@PRANEETH-ALURU: Thank you for bringing this up.

@BruceWen120: Could you please let us know if there are any updates on this thread?

**Comment by phaniparsa** [comment]
@BruceWen120 and @PRANEETH-ALURU : After some brainstorming within our team, one of our team members pointed out that there are 22555 unique labels in the training csv file posted in Kaggle. Maybe, this is the reason the output layers in the LSTM and LSTM-SA models have 22555 neurons in the final layer. Here is the Kaggle link for the dataset: https://www.kaggle.com/xhlulu/medal-emnlp . Hope this helps. Thanks!

**Comment by BruceWen120** [comment]
Thank you @phaniparsa and sorry that I didn't provide an update.

Your answers seem right. As shown in the paper, 24005 is the total number of abbreviations from ADAM. We used that number as the output dimension for Electra, i.e. one per each abbreviation. On the other hands, there seem to be 22555 unique abbreviations in the data (some abbreviations don't appear), and when training LSTM and LSTM-SA, an earlier version of the code seemed to use that number as the output dimension.

I hope this helps.

**Comment by phaniparsa** [comment]
@BruceWen120 : Yes this helps. However I have a follow-up question. Is the same data used to train both LSTM and Electra models? If yes, any thoughts on whether it is wise to add in more classes in the output layer, while the training data does not contain any samples from these classes? Or was a different dataset used to train the electra model?

**Comment by BruceWen120** [comment]
Yes they are trained on the same dataset.

I would say they should be equivalent or similar, because even if you add more classes their corresponding parameters in the output layer are not being trained (since you don't have positive examples).

**Comment by phaniparsa** [comment]
@BruceWen120 : Thank you very much, this helps (y)

**Comment by BruceWen120** [comment]
I will close this issue now. Feel free to reopen if you have new questions.
